Sprint-1/exercises/decimal.js

```diff
@@ -1,4 +1,24 @@
+// Create a variable called wholeNumberPart and assign to it an expression that evaluates to 56 ( the whole number part of num )
+function wholeNumberPart(inputNumber) {
+    return Math.trunc(inputNumber);
+}
+
+// Create a variable called decimalPart and assign to it an expression that evaluates to 0.5678 ( the decimal part of num )
+function decimalPart(inputNumber) {
```

Comment:
This is a clever method but it won't work properly when inputNumber is `0.000000002`.

Reply:
My original version of this function was:
`return Math.abs(inputNumber - wholeNumberPart(inputNumber));`

but it gave the result:
`decimalPart from 56.5678 = 0.56779999999999983`

which in my opinion in the context of the problem statement is not correct

if we take your example with `0.000000002`, then my version, and I think you hinted at my original version, both options will give the result but with a number in the exponent format. To do everything "nicely" you need to write a full-fledged conversion function from the "exponent". But there is one "but", if the bit depth is more than 15, then you need to connect, for example, the `decimal.js` library
In **Python**, I use a similar library for such "tricky" problems. I suggest limiting ourselves to this comment here because I demonstrated an understanding of the problem and the way to solve it.

Reply:
`Math.abs(inputNumber - wholeNumberPart(inputNumber))`, even if it yield 0.56779999999999983, is perfectly 
normal and acceptable solution. That's just the nature of floating point arithmetic. You will probably get the same result in any language that uses double precision floating point number. 

If we want to show only 4 decimal places, we can "format" the output to have only 4 decimal places without changing the value of the variable. 

However, if we were to use an approach that depends on converting the number to a String first, we need to be aware how number type is converted to string. In the example I showed you, the number is converted to a string in scientific format and as a result your code won't work properly.

Reply:
The whole point is that any solution path will be converted to scientific format if it is an exponent. The conversion occurs when the result is outputting to the console. Most likely, a background conversion to a "string" occurs, that's why. I tried to explain this in my comment, but apparently, I didn't do it very cleanly.
example:
`decimalPart from 56.5678 = 0.567799999999999983`
was given to explain why I converted to a string right away in my solution. There is no doubt that this is normal from a mathematical point of view, but I tried to make a "universal" visualisation, forgetting that the exponent is visualised in scientific format by default in many programming languages

For numbers like `0.000000002`, you can play around with methods `.toFixed()` and/or `.toPrecision()` not forgetting about `.substring()`
But if such a number has more than 14 digits we get a "cut" number to an integer, after that we need to connect a library, for example `decimal .js`, as I wrote above.

That is, I wanted to say that I could not find a solution for outputting the fractional part for numbers like `0.000000002` without writing a conversion function from scientific format.

I came to this conclusion after a series of studies and experiments. Perhaps I did them wrong.

Shortly, I tried to get a string from the float and cut off only the decimal part, forgetting numbers like `0.000000002`. If I had remembered it, I would have written a solution that performs the conversion from scientific format. But then we still run into the float limitation of no more than 17 decimal places. Frankly, I did not dig further to see if it was possible to get a "longer" float. On the stack I worked on before, I had a float limitation of 26 decimal places. If I understood correctly, then JavaScript has a limitation of 17.